src/ChartInternal/shape/polar.ts

```diff
@@ -0,0 +1,267 @@
+import {select as d3Select} from "d3-selection";
+import {arc as d3Arc, pie as d3Pie} from "d3-shape";
+import CLASS from "../../config/classes";
+import {getRange} from "../../module/util";
+
+export default {
+	initPolar(): void {
+		const $$ = this;
+		const {$el} = $$;
+
+		$el.polar = $el.main.select(`.${CLASS.chart}`).append("g")
+			.attr("class", CLASS.chartPolars);
+
+		// level
+		$el.polar.levels = $el.polar.append("g")
+			.attr("class", CLASS.levels);
+
+		// arcs
+		$el.polar.arcs = $el.polar.append("g")
+			.attr("class", CLASS.chartPolarArcs);
+
+		// Let each value be 1, thus every arc has same central angle
+		// To match central angle with specific data, change "1" to specific function.
+		$$.polarPie = d3Pie()
+			.value(1);
+	},
+
+	getPolarSize(): [number, number] {
+		const $$ = this;
+		const {state: {arcWidth, arcHeight}} = $$;
+		const size = (Math.min(arcWidth, arcHeight) - 10) / 2;
+
+		return [size, size];
+	},
+
+	getPolarArc(d): string {
+		const $$ = this;
+		const {config} = $$;
+		const [width, height] = $$.getPolarSize();
+		const radius = config.polar_size_ratio * Math.min(width, height);
+		const ceilDataMax = $$.getMinMaxData().max[0].value;
+		const dataMax = config.polar_level_max || ceilDataMax;
+
+		// TODO: remove magic number
+		const innerRadius = config.polar_size_ratio * config.polar_padding * 0.75;
+
+		return d3Arc()
+			.innerRadius((d: any) => innerRadius * d.data.values.reduce((a, b) => a + b.value, 0) / dataMax)
+			.outerRadius((d: any) => d.data.values.reduce((a, b) => a + b.value, 0) / dataMax * radius)(d) || "M 0 0";
+	},
+
+	updateTargetsForPolarArc(): void {
+		// stay empty
+	},
+
+	redrawPolar(): void {
+		const $$ = this;
+		const {config} = $$;
+		const {polar} = $$.$el;
+		const translate = $$.getTranslate("polar");
+		const classChartArc = $$.getChartClass("Arc");
+		const hasInteraction = config.interaction_enabled;
+		const startingAngle = config.polar_startingAngle || 0;
+		const padding = config.polar_padding;
+		const padAngle = (
+			padding ? padding * 0.01 : config.polar_padAngle
+		) || 0;
+
+		polar.attr("transform", translate);
+		$$.updatePolarLevel();
+
+		let polarPie = $$.polarPie;
+
+		polarPie = polarPie
+			.startAngle(startingAngle)
+			.endAngle(startingAngle + (2 * Math.PI))
+			.padAngle(padAngle);
+
+		// remove all `g` tag from previous polar chart
+		polar.arcs.selectAll("g").data([])
```

Comment:
IMO isn't good approach on every redraw, because removing all nodes and re appending isn't cost effective. 
It should be only updating `d` attributes to reduce unnecessary node handling.

![polar2](https://user-images.githubusercontent.com/2178435/148359977-8b5c7051-9172-433c-a9bd-2e6f2b705be5.gif)